components/dialog/test/dialog-mixin.vdiff.js

```diff
@@ -51,6 +76,24 @@ describe('dialog-mixin', () => {
 					});
 				});
 			});
+
+			it('delayed-content', async() => {
+				const el = await fixture(
+					`<d2l-dialog title-text="Delayed Dialog"><${delayedTag}></${delayedTag}></d2l-dialog>`,
+					{
+						awaitLoadingComplete: false
+					}
+				);
+				await waitUntil(() => el.querySelector(delayedTag) !== null);
+				const delayedEl = el.querySelector(delayedTag);
+				await delayedEl.updateComplete;
+
+				setTimeout(() => el.opened = true);
+				setTimeout(() => delayedEl.finishLoading(), 100);
+				await oneEvent(el, 'd2l-dialog-open');
```

Comment:
The way this is going to work right now:
1. Dialog opens, starts its opening logic
2. Once the opening logic is done and the initial resize happens, the event will fire
3. Screenshot is taken
4. Sometime later, the inner timeout fires which causes the delayed element to finish loading -- but it's too late

Once this is fixed, the sequence will be:
1. Dialog opens, starts its opening logic
2. It's going to wait for everything inside it to be "loading complete"
3. The inner timeout fires which causes the delayed element to finish loading
4. Resize happens, the event will fire
5. Screenshot is taken